### DISCUSSION TITLE
how does test_labels and train_labels folders containing black images as labels.

### DISCUSSION THREAD

**Discussion by jayajkd** [thomasfermi/Algorithms-for-Automated-Driving#34] [General]
I am working on lane detection, and I have a doubt about the test_labels and train_labels folders containing black images as labels. Could you please explain how these black images were generated? If possible, could you provide some code examples?

**Comment by thomasfermi** [comment]
Hi @jayajkd! The label images are black if you look at them with an Image Viewer. But if you open them with python you will see that the pixels have values of 0,1 or 2, indicating whether they are nothing, left lane boundary or right lane boundary. Since 1 and 2 are very small compared to 255, this looks like 0, i.e. black in an Image Viewer. 

The images were generated using this program:
code/solutions/lane_detection/collect_data.py

**Comment by jayajkd** [comment]
Thanks for your reply

**Comment by jayajkd** [comment]
And  one more thing i need to confrom it s must use carla simulatin in system to run this collect_data.py right..

**Comment by jayajkd** [comment]
And one more i need to clarify is  must install carla in  my laptop to  run
collect_data.py right..

[image: Mailtrack]
<https://mailtrack.io?utm_source=gmail&utm_medium=signature&utm_campaign=signaturevirality11&>
Sender
notified by
Mailtrack
<https://mailtrack.io?utm_source=gmail&utm_medium=signature&utm_campaign=signaturevirality11&>
09/20/23,
10:32:27 PM

On Wed, Sep 20, 2023 at 8:38 PM Mario Theers ***@***.***>
wrote:

> Closed #34
> <https://github.com/thomasfermi/Algorithms-for-Automated-Driving/discussions/34>
> as resolved.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/thomasfermi/Algorithms-for-Automated-Driving/discussions/34>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/A2CHPK3JWYVYNEKPNHGZQLTX3MBIHANCNFSM6AAAAAA47JUPXQ>
> .
> You are receiving this because you were mentioned.Message ID:
> <thomasfermi/Algorithms-for-Automated-Driving/repo-discussions/34/discussion_event/868228
> @github.com>
>


-- 

*----------------------------------------------*
*Regards:*
*Dontabhaktuni Jayakumar, M.Tech, (PhD)*
*PhD Research Scholar, *

*VIT-AP,Amaravathi,*
*Vijayawada. *
*Mob: 90320 90461.*

**Comment by thomasfermi** [comment]
Yes, that is true. In the appendix there is an explanation regarding Carla installation:
https://thomasfermi.github.io/Algorithms-for-Automated-Driving/Appendix/CarlaInstallation.html

But note that you do not need to collect that data yourself, you can also just download the data I collected.
